src/io/file_descriptor_io.cr

```diff
@@ -258,6 +259,8 @@ class FileDescriptorIO
     if event = @event
       event.free
     end
+
+    raise err if err
```

Comment:
If there is an error trying to close the file, should we really mark the file as closed?